SPECS/grub2/grub2.spec

```diff
@@ -118,6 +122,7 @@ Unsigned GRUB UEFI image
 %package efi-binary
 Summary:        GRUB UEFI image
 Group:          System Environment/Base
+Requires:       grub2
```

Comment:
I was thinking we would eventually separate out just the grub2-mkconfig script + the /etc/grub.d/* configs into a separate %{name}-mkconfig package. That way, an image that is natively building its config using grub2-mkconfig will always have it available. I was planning to do that in the second PR when we switch over, but since this is an image space concern I will do it in this one.

Reply:
Done, see latest specfiles which Requires: grub2-mkconfig.

Reply:
Why do we need to require any tooling by default? This is also introducing a functional dependency change in current package behavior